src/modules/graphics/vulkan/Graphics.cpp

```diff
@@ -1033,11 +1003,22 @@ void Graphics::setScissor()
 
 	states.back().scissor = false;
 
-	VkRect2D scissor{};
-	scissor.offset = { 0, 0 };
-	scissor.extent = swapChainExtent;
+	if (renderPassState.active)
+	{
+		VkRect2D scissor{};
+		scissor.offset = { 0, 0 };
+		if (renderPassState.isWindow)
+			scissor.extent = swapChainExtent;
+		else
+		{
+			double dpiScale = getCurrentDPIScale();
+
+			scissor.extent.width = static_cast<double>(renderPassState.width) * dpiScale;
```

Comment:
I think renderPassState.width/height are already in pixels, so they shouldn't need the dpiScale multiplier.